### PR TITLE
[bug#10696] Make lazyLines fully lazy

### PR DESCRIPTION
**Pull request by NthPortal**
Fixes scala/bug#10696

### CODE REVIEW

**Comment by SethTisue** [comment]
can/should this have a test?

**Comment by NthPortal** [comment]
Probably, but I'm not really sure how

**Comment by SethTisue** [comment]
```
scala> sys.process.Process("false").lineStream
                                    ^
       warning: method lineStream in trait ProcessBuilder is deprecated (since 2.13.0): use lazyLines
java.lang.RuntimeException: Nonzero exit code: 1
  at scala.sys.package$.error(package.scala:26)
  at scala.sys.process.BasicIO$Streamed$.next$1(BasicIO.scala:68)
  at scala.sys.process.BasicIO$Streamed$.$anonfun$apply$7(BasicIO.scala:71)
  at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lineStream(ProcessBuilderImpl.scala:175)
  at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lineStream(ProcessBuilderImpl.scala:120)
  ... 33 elided

scala> sys.process.Process("false").lazyLines
res11: LazyList[String] = LazyList(?)
```

?

**Comment by NthPortal** [comment]
I... forgot `false` existed. Thanks!

File: test/junit/scala/sys/process/ProcessTest.scala [hunk 1]
```diff
@@ -33,4 +35,15 @@ class ProcessTest {
     val res = Process("cat", tempFiles.map(_.getAbsolutePath)).!
     assertEquals(0, res)
   }
+
+  @Test def t10696(): Unit = {
```
Comment (NthPortal) [inline test/junit/scala/sys/process/ProcessTest.scala:39]: this should probably use `testily`

**Review by julienrf — APPROVED** [review]

**Comment by som-snytt** [comment]
I'm looking at a different Process bug, and a test here would have helped. I agree constructing tests for this codebase is challenging.

**Comment by NthPortal** [comment]
What in particular did the test I added not cover?

**Comment by som-snytt** [comment]
Oh, I just saw what you meant by the test you added. So I'm working on what if a process doesn't start successfully, not if it starts but exits non-zero.

File: src/library/scala/sys/process/ProcessBuilderImpl.scala [hunk 2]
```diff
@@ -155,11 +155,11 @@ private[process] trait ProcessBuilderImpl {
...
       capacity: Integer
     ): LazyList[String] = {
-      val streamed = Streamed[String](nonZeroException, capacity)
-      val process  = run(BasicIO(withInput, streamed.process, log))
+      val lazilyListed = LazilyListed[String](nonZeroException, capacity)
+      val process      = run(BasicIO(withInput, lazilyListed.process, log))
 
-      Spawn(streamed done process.exitValue())
-      streamed.stream()
+      Spawn(lazilyListed done process.exitValue())
```
Comment (som-snytt) [inline src/library/scala/sys/process/ProcessBuilderImpl.scala:161]: If `exitValue` throws, nothing unblocks the caller who forces the lazy list.

**Comment by NthPortal** [comment]
#7362 addresses some problems with this PR

**Pull request merged**